### PR TITLE
[SDLNativeRules] - Defect : RetvalIgnoredFuncCouldFail

### PR DESCRIPTION
**Pull request by ericwol-msft**
# Checklist
- [x] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-c/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [x] This pull-request is submitted against the `master` branch. 
  - [x] I have merged the latest `master` branch prior to submission and re-merged as needed after I took any feedback.
  - [x] I have squashed my changes into one with a clear description of the change.

# Description of the problem
Details :
Return value ignored: 'snprintf'. 
at d:\a\1\s\iothub_client\samples\iothub_client_sample_upload_to_blob_mb\iothub_client_sample_upload_to_blob_mb.c@53,24

### CODE REVIEW

File: iothub_client/samples/iothub_client_sample_upload_to_blob_mb/iothub_client_sample_upload_to_blob_mb.c [hunk 1]
```diff
@@ -50,7 +50,11 @@ static IOTHUB_CLIENT_FILE_UPLOAD_GET_DATA_RESULT getDataCallback(IOTHUB_CLIENT_F
 
             if (block_count < 100)
             {
-                snprintf(data_to_upload, sizeof(data_to_upload), data_to_upload_format, block_count);
+                if (snprintf(data_to_upload, sizeof(data_to_upload), data_to_upload_format, block_count) < 0)
```
Comment (danewalton) [inline iothub_client/samples/iothub_client_sample_upload_to_blob_mb/iothub_client_sample_upload_to_blob_mb.c:53]: should there also be a check that the written size < sizeof(data_to_upload)?

**Review by danewalton — APPROVED** [review]

File: iothub_client/samples/iothub_client_sample_upload_to_blob_mb/iothub_client_sample_upload_to_blob_mb.c (hunk 1 above)
Comment (ericwol-msft) [reply iothub_client/samples/iothub_client_sample_upload_to_blob_mb/iothub_client_sample_upload_to_blob_mb.c:53]: ?? 
snprintf will return less than zero if the written size (including null) is greater than sizeof(data_to_upload)
size is computed from strlen()(
??

Comment (danewalton) [reply iothub_client/samples/iothub_client_sample_upload_to_blob_mb/iothub_client_sample_upload_to_blob_mb.c:53]: From here: https://www.cplusplus.com/reference/cstdio/snprintf/ (emphasis mine)

> The number of characters that would have been written if n had been sufficiently large, not counting the terminating null character.
If an encoding error occurs, a negative number is returned.
Notice that only when this returned value is **non-negative and less than n**, the string has been completely written.

If it's more than n then it wasn't completely written?

Comment (ericwol-msft) [reply iothub_client/samples/iothub_client_sample_upload_to_blob_mb/iothub_client_sample_upload_to_blob_mb.c]: fixed

**Pull request merged**